### PR TITLE
Fixes issue #3552: Mention Table interface in "Creating a New Table File" in FITS docs

### PR DESCRIPTION
**Pull request by vinayak-mehta**
#3552

### CODE REVIEW

**Comment by embray** [comment]
Thanks, this seems like a good start.  Though I would probably write this differently, ultimately.  I would want to move it closer to the top of the section and go through a more complete example (also with links to the Table docs).  Perhaps showing how to create the same table using both the Table interface, and the old lower-level PyFITS interface, and also an explanation as to why the former should be preferred in most cases.

**Comment by vortex-ape** [comment]
@embray - Please point out the mistakes and help me refine it.

**Comment by embray** [comment]
@vortex-ape Thanks, this is looking good.  I don't think there's much in the way of "mistakes" though I have a few other comments.  I'll get back to it in a bit.  But this is definitely on the right track for what I'm looking for.

**Comment by embray** [comment]
Thanks again for this, but I'm probably going to finish up this update to the docs myself, though I will include your example and make sure you're credited.  I just want to go a little more in-depth, and actually refactor the docs altogether to emphasize this, and other simpler options for creating table HDUs.  This is perfectly good work, but it's only a start.

**Comment by hamogu** [comment]
current fits maintaienrs @embray @MSeifert04 @pllim : I'm just looking through old, open PRs. Is there any reason this has not been merged yet? Yes, somebody could rework the docs more fully but that has not happened for the last two years, so I suggest merging this as is.
(The Travis failure seems unrelated, but in any case, somebody with the right priviliges should restart Travis, since that run was about 2 years ago.)

File: docs/io/fits/index.rst [hunk 1]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
 extension HDU, not a primary. There are two kinds of FITS table extensions:
 ASCII and binary. We'll use binary table examples here.
 
+.. note::
+    
+    If you want to create a simple binary FITS table with no other HDUs,
+    you should use the Table_ interface in Astropy instead and then
```
Comment (pllim) [inline docs/io/fits/index.rst]: It is simpler to just do
```
:class:`~astropy.table.Table`
```
This way, "stable" will properly link to the API in "stable", "latest" to "latest", and so on.

File: docs/io/fits/index.rst [hunk 2]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
 ASCII and binary. We'll use binary table examples here.
 
+.. note::
+    
+    If you want to create a simple binary FITS table with no other HDUs,
+    you should use the Table_ interface in Astropy instead and then 
+    write to FITS.
+
+    >>> from astropy.table import Table
+    >>> t = Table([[1,2],[4,5],[7,8]], names=('a','b','c'))
```
Comment (pllim) [inline docs/io/fits/index.rst]: Wouldn't hurt to follow PEP8 here and add spaces after commas.

File: docs/io/fits/index.rst [hunk 3]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
+    >>> c3 = np.array([7,8])
+    >>> t = pyfits.BinTableHDU.from_columns([
+    ...     pyfits.Column(name='a', array=c1),
+    ...     pyfits.Column(name='b', array=c2),
+    ...     pyfits.Column(name='c', array=c3)])
+    >>> t.writeto('table.fits')
+
+    When using "lower-level" FITS interface, it is not usually necessary to
+    create column object explicitly as long as the data is stored in a 
+    `structured array <http://docs.scipy.org/doc/numpy/user/basics.rec.html>`_.
```
Comment (pllim) [inline docs/io/fits/index.rst]: `https` now.

File: docs/io/fits/index.rst [hunk 4]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
+    `structured array <http://docs.scipy.org/doc/numpy/user/basics.rec.html>`_. 
+    
+    The data in the structured numpy array can directly be used to create a FITS table. 
+    Use of old interface is required only in certain cases.
+    
+    Benefits:
+
+    - Code complexity is decreased.
+
+.. _Table: http://astropy.readthedocs.org/en/latest/api/astropy.table.Table.html#astropy.table.Table
```
Comment (pllim) [inline docs/io/fits/index.rst]: This can be removed as per my suggestion above.

File: docs/io/fits/index.rst [hunk 5]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
+    When using "lower-level" FITS interface, it is not usually necessary to
+    create column object explicitly as long as the data is stored in a 
+    `structured array <http://docs.scipy.org/doc/numpy/user/basics.rec.html>`_. 
+    
+    The data in the structured numpy array can directly be used to create a FITS table. 
+    Use of old interface is required only in certain cases.
+    
+    Benefits:
+
+    - Code complexity is decreased.
```
Comment (pllim) [inline docs/io/fits/index.rst]: Is it really necessary to make a bulleted list out of a single item like this?

File: docs/io/fits/index.rst [hunk 6]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
+    
+    If you want to create a simple binary FITS table with no other HDUs,
+    you should use the Table_ interface in Astropy instead and then 
+    write to FITS.
+
+    >>> from astropy.table import Table
+    >>> t = Table([[1,2],[4,5],[7,8]], names=('a','b','c'))
+    >>> t.write('table.fits', format='fits')
+
+    Same code in PyFITS would look like this:
```
Comment (MSeifert04) [inline docs/io/fits/index.rst]: PyFITS? 😄

Comment (pllim) [inline docs/io/fits/index.rst]: Now that you mention it, PyFITS move to `astropy.io.fits` is so long ago now that we can probably replace PyFITS reference with `astropy.io.fits`. :thinking:

Comment (MSeifert04) [inline docs/io/fits/index.rst]: no "probably" about it. Please change it to `astropy.io.fits`. PyFITS has been deprecated for 3 years now (early 2014).

Comment (bsipocz) [inline docs/io/fits/index.rst]: And I vaguely remember previous PRs that got rid of similar mentions of pyfits both in code and docs.

File: docs/io/fits/index.rst [hunk 7]
```diff
@@ -473,6 +473,42 @@ table's structure needs more information. First of all, tables can only be an
...
+
+    >>> import pyfits
+    >>> import numpy as np
+    >>> c1 = np.array([1,2])
+    >>> c2 = np.array([4,5])
+    >>> c3 = np.array([7,8])
+    >>> t = pyfits.BinTableHDU.from_columns([
+    ...     pyfits.Column(name='a', array=c1),
+    ...     pyfits.Column(name='b', array=c2),
+    ...     pyfits.Column(name='c', array=c3)])
```
Comment (saimn) [inline docs/io/fits/index.rst]: This doesn't work, a format is required, something like `fits.Column(name='a', array=[1,2], format='K')`.

**Comment by pllim** [comment]
@SaraOgaz , are you going to Katniss this one too? Or should I take over? :joy:

**Comment by SaraOgaz** [comment]
@pllim  depends on how early you want it, heh.  Do we know that @vinayak-mehta won't get to it?  Are they still around?

**Comment by pllim** [comment]
I think code freeze for v2 is May-ish, so we have some time yet.

**Comment by pllim** [comment]
But if you want people to know about this in `latest` doc, then ASAP would be nice. Not sure if @vinayak-mehta is still interested. This PR is 2 years old. He might have moved on to something else.

**Comment by SaraOgaz** [comment]
In that case, sure.  I'll try and get to that today. Might need some guidance to make sure @vinayak-mehta still get's the credit.

**Comment by pllim** [comment]
Here it is, Katniss: http://stackoverflow.com/questions/3042437/change-commit-author-at-one-specific-commit

**Comment by bsipocz** [comment]
@pllim @SaraOgaz - I always feel weird about these history hacks and credit amendments. What about cherry picking the original commit from here on top of the current master, and do the fixes in a separate commits? That case the first commit will be credited to @vinayak-mehta and will have the original date, too without any hacking involved. Any additional fix will be credited to the person taking up the stab and finishing the issue.

**Comment by pllim** [comment]
@bsipocz , I never done that before but I like the suggestion. What are the incantation involved? Thanks!

**Comment by saimn** [comment]
You can get a branch from this PR and continue to work on it (possibly squashing the commits in one):
```
git fetch upstream pull/3558/head:pr-3558
git checkout pr-3558
```

**Comment by bsipocz** [comment]
@pllim - The other option, start with a new branch on top of the current master, and pick the 3 commits from here. ``git cherry-pick 39bc597edf80d25b77578c8b079b2c55d739d9e3 20b3dc05152b333650b67643a5f8d08f82a53582 335c72ccaae180b661b7cc5c34370c2e0755a72e`` I agree with @saimn, that it would be then best to squash them together.

**Comment by bsipocz** [comment]
hmm, in my solution you may need to add the fork of @vinayak-mehta as a remote to be able to see these commits (I usually have the forks of many people as remotes, so haven't tried this without it).

**Comment by pllim** [comment]
I took over and moved it to #5893 . This will close when that one is merged. FYI.

**Comment by hamogu** [comment]
Thanks to everybody who contributed work on this PR and the discussion! 

You might hear from me again: I've decided to comb through all open PRs to find those that are still open while  "almost but not quite finished". Some are really good ideas, but the longer we wait, the more extra work somebody has to put in to finish them (rebase etc.). So, I'll try to call out those that seem easy to finish (or do it myself if I know enough about the relevant sub-package). Also, the number of open PRs can be intimidating to newcomers.

**Comment by vinayak-mehta** [comment]
Sorry for not being able to address this, saw it just today.

**Comment by pllim** [comment]
@vinayak-mehta , no worries!

**Pull request closed without merging**